src/commands/mod.rs

```diff
@@ -0,0 +1,39 @@
+use clap::{ArgEnum, Parser};
+use std::fmt::{Display, Formatter};
+
+#[derive(Parser)]
+pub struct Input {
+    #[clap(arg_enum)]
+    pub cmd: CargoCommand,
+    pub cargo_opts: Vec<String>,
+}
+
+#[derive(ArgEnum, Copy, Clone, Ord, PartialOrd, Eq, PartialEq)]
```

Comment:
minor nit: from reading [the docs](https://docs.rs/clap/latest/clap/derive.ArgEnum.html) it looks like this is deprecated in favor of `ValueEnum` ? From what I can tell they are functionally the same though

Reply:
I think with the deprecation it's almost a renaming but I can change that. From my understanding ArgEnum derives ValueEnum but I haven't looked at the source of Clap to confirm.